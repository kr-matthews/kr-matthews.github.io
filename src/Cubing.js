function Cubing(props) {
  return (
    <div className="page narrow-page">
      <h1>Cubing</h1>
      <p>WIP</p>
    </div>
  );
}

export default Cubing;
