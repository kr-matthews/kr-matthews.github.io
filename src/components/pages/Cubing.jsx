import Link from "../common/Link";

export default function Cubing() {
  return (
    <div className="page narrow-page">
      <h1>Cubing</h1>
      <p>
        I know how to solve a Rubik's Cube (blindfolded), and since 2010{" "}
        <Link
          href="https://www.worldcubeassociation.org/persons/2010MATT02"
          isExternal
        >
          I've competed
        </Link>{" "}
        in over 100 competitions across North America, many of which I helped
        organize and run as a volunteer -- including the{" "}
        <Link href="https://nac2022.speedcubingcanada.org/en" isExternal>
          inaugural North American Championship
        </Link>{" "}
        in 2022 in Toronto with 500 competitors over 4 days. I've held a few
        Canadian records in the multiple-blindfolded event and the fewest moves
        event.
      </p>
      <p>
        I've been a volunteer member of the World Cube Association's{" "}
        <Link href="https://github.com/thewca" isExternal>
          software team
        </Link>{" "}
        for over a year now, where my focus is on transitioning the front-end of
        the website from Rails to React. If you can code, consider volunteering,
        as they always need more help. The website is run mainly with Ruby on
        Rails and React.
      </p>
      <p>
        Reconstructions of some of my notable solves can be found{" "}
        <Link
          href="https://docs.google.com/spreadsheets/d/1PJ1k_EXqv_mafcMDRbrD5OAyBHie2z5PH5OkHLs8BtE/edit#gid=1878622909"
          isExternal
        >
          here
        </Link>
        , including both regular and blindfolded solving for several different
        events. Note that you can apply existing filters to filter by event. For
        my fewest moves solves, see{" "}
        <Link
          href="https://csclub.uwaterloo.ca/~krmatthe/FMC/old-index"
          isExternal
        >
          here
        </Link>
        , but note that this is not currently up-to-date and I have plans for
        creating a more comprehensive spreadsheet in the future (which I may or
        may not ever finish).
      </p>

      <h3>Competitions, World Cube Association, and Speedcubing Canada</h3>

      <p>
        If you're interested in attending a competition,{" "}
        <Link
          href="https://www.worldcubeassociation.org/competitions"
          isExternal
        >
          find one near you
        </Link>{" "}
        on the{" "}
        <Link href="https://www.worldcubeassociation.org/" isExternal>
          World Cube Association
        </Link>{" "}
        (WCA) -- or, if you're in Canada, sign up for{" "}
        <Link href="https://www.speedcubingcanada.org/" isExternal>
          Speedcubing Canada
        </Link>
        's (SCC){" "}
        <Link
          href="https://share.hsforms.com/1eWkfr6anSF-HKTZJVrElZQcrqzz"
          isExternal
        >
          mailing list
        </Link>{" "}
        to be notified of new competitions. The WCA is the governing body of
        Rubik's Cube competitions and SCC is Canada's regional organization
        which is recognized by the WCA. Both are not-for-profits run purely by
        volunteers. Note that you don't need to be competitive/fast to attend a
        competition; everyone is welcome.
      </p>
      <p>
        If you're interested in organizing a competition, I have a{" "}
        <Link
          href="https://krmatthews.notion.site/Competition-Template-c2e4edd77fe74cc0a6498c1e68069bda"
          isExternal
        >
          template
        </Link>{" "}
        I made for organizers here in BC, Canada. I would, however, recommend
        attending at least a few competitions before organizing one, and
        thoroughly reading through the template to make sure the expectations
        and workload are clear.
      </p>

      <h3>Software and Tools</h3>

      <p>
        {" "}
        I have several <Link to="/projects">web apps</Link> based on cubes,
        primarily designed to help with learning and practising blindfolded
        solving -- and I intend to create more in the future. If you've recently
        learned to solve a cube blindfolded, or struggle with mistakes while
        practising, you'll find them helpful.
      </p>
      <p>
        There are plenty of other tools out there, made by other people. I don't
        have a comprehensive list handy, but you can search for them.
      </p>

      <h3>Personal Blindfolded-Solving System</h3>

      <p>
        I have{" "}
        <Link
          href="https://docs.google.com/spreadsheets/d/1Ml7Jl5Yeb9HeE14xO7j1Nc99sw08oiBjLbi_nYJk9rU/edit"
          isExternal
        >
          a spreadsheet
        </Link>{" "}
        which lists the images I use for memorizing cubes and the move sequences
        (also known as algorithms in the community) which I use to solve each
        pair of pieces. (I also use the images for memory sports, including
        memorizing numbers and playing cards.) There are many online resources
        for these in general, including{" "}
        <Link href="https://bestsiteever.ru/tables/" isExternal>
          this list of algorithm lists
        </Link>{" "}
        and{" "}
        <Link
          href="https://docs.google.com/spreadsheets/d/1Fi4xgUz5b23UXMlHq7Tt5C8Ak8-U3XdbeQ9Anw68BQc/edit#gid=0"
          isExternal
        >
          this collection of letter pair images
        </Link>
        .
      </p>
      <p>
        My system is relatively unique in the blindfolded solving community in
        that it is based on phonemes and pronunciation rather than letters and
        spelling. Memory sports athletes usually use phonemes, as far as I
        understand, though I came up with the idea independently. It took
        literally years of experimenting to get it to where it is today.
      </p>

      <h3>Method Development: Weak Swap</h3>

      <p>
        Together with{" "}
        <Link
          href="https://www.worldcubeassociation.org/persons/2016SIGG01"
          isExternal
        >
          Graham Siggins
        </Link>{" "}
        I developed a slight optimization for blindfolded solving called Weak
        Swap. Graham pointed out a special case to me one day where he was able
        to reduce his algorithm count by one. I played around and realized that
        it could be generalized to apply to a large portion of scrambles, and
        together we nailed down what turned out to be a relatively simple method
        to incorporate it into every solve when applicable. Graham has a{" "}
        <Link
          href="https://www.youtube.com/watch?v=MyeQkcsAzUE&ab_channel=GrahamSiggins"
          isExternal
        >
          sequence of videos
        </Link>{" "}
        on the topic, and the calculations I did can be found on{" "}
        <Link href="https://github.com/kr-matthews/weak-swap" isExternal>
          GitHub
        </Link>
        . I intend to eventually write a blog post about it.
      </p>
    </div>
  );
}
