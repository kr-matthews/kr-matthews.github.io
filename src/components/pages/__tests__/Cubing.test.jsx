import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Cubing from "../Cubing";
import { BrowserRouter } from "react-router-dom";

describe("components/pages/Cubing", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Cubing />, {
      wrapper: ({ children }) => <BrowserRouter>{children}</BrowserRouter>,
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
