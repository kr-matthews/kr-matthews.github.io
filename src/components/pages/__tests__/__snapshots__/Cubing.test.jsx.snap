// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Cubing matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-rzly3e"
  >
    <div
      class="css-jydvbh"
    >
      <h1>
        Cubing
      </h1>
      <p
        style="margin-bottom: 2em;"
      >
        I know how to solve a Rubik's Cube (blindfolded), and since 2010 
        <a
          class="chakra-link css-1108m31"
          href="https://www.worldcubeassociation.org/persons/2010MATT02"
          rel="noopener"
          target="_blank"
          to=""
        >
          I've competed
        </a>
         in over 125 competitions across North America, many of which I helped organize and run as a volunteer -- including the 
        <a
          class="chakra-link css-1108m31"
          href="https://nac2022.speedcubingcanada.org/en"
          rel="noopener"
          target="_blank"
          to=""
        >
          inaugural North American Championship
        </a>
         in 2022 in Toronto with 500 competitors over 4 days. I've held a few Canadian records in the multiple-blindfolded event and the fewest moves event.
      </p>
      <div
        class="chakra-tabs css-0"
      >
        <div
          aria-orientation="horizontal"
          class="chakra-tabs__tablist css-k008qs"
          role="tablist"
        >
          <button
            aria-controls="tabs-:r0:--tabpanel-0"
            aria-selected="true"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="0"
            id="tabs-:r0:--tab-0"
            role="tab"
            tabindex="0"
            type="button"
          >
            Competitions
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-1"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="1"
            id="tabs-:r0:--tab-1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Software
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-2"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="2"
            id="tabs-:r0:--tab-2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Reconstructions
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-3"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="3"
            id="tabs-:r0:--tab-3"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Blindfolded
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-4"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="4"
            id="tabs-:r0:--tab-4"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Weak Swap
          </button>
        </div>
        <div
          class="chakra-tabs__tab-panels css-8atqhb"
        >
          <div
            aria-labelledby="tabs-:r0:--tab-0"
            class="chakra-tabs__tab-panel css-sjt5nk"
            id="tabs-:r0:--tabpanel-0"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              The 
              <a
                class="chakra-link css-1108m31"
                href="https://www.worldcubeassociation.org/"
                rel="noopener"
                target="_blank"
                to=""
              >
                World Cube Association
              </a>
               (WCA) is the governing body of Rubik's Cube competitions. 
              <a
                class="chakra-link css-1108m31"
                href="https://www.speedcubingcanada.org/"
                rel="noopener"
                target="_blank"
                to=""
              >
                Speedcubing Canada
              </a>
               (SCC) is Canada's regional organization which is recognized by the WCA. Both are not-for-profits run by volunteers.
            </p>
            <p>
              If you're interested in attending a competition, you can 
              <a
                class="chakra-link css-1108m31"
                href="https://www.worldcubeassociation.org/competitions"
                rel="noopener"
                target="_blank"
                to=""
              >
                find one near you
              </a>
               on the WCA. Alternatively, if you're in Canada, sign up for SCC's 
              <a
                class="chakra-link css-1108m31"
                href="https://share.hsforms.com/1eWkfr6anSF-HKTZJVrElZQcrqzz"
                rel="noopener"
                target="_blank"
                to=""
              >
                mailing list
              </a>
               to be notified of new competitions in your province. Note that you don't need to be competitive or fast to attend a competition; everyone is welcome.
            </p>
            <p>
              If you're interested in organizing a competition, I have a 
              <a
                class="chakra-link css-1108m31"
                href="https://krmatthews.notion.site/Competition-Template-c2e4edd77fe74cc0a6498c1e68069bda"
                rel="noopener"
                target="_blank"
                to=""
              >
                template
              </a>
               I made for organizers here in BC, Canada. I would, however, recommend attending at least a few competitions before organizing one, and thoroughly reading through the template to make sure the expectations and workload are clear.
            </p>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-1"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-1"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              I've been a volunteer member of the World Cube Association's 
              <a
                class="chakra-link css-1108m31"
                href="https://github.com/thewca"
                rel="noopener"
                target="_blank"
                to=""
              >
                software team
              </a>
               since 2021, where my focus is on transitioning the front-end of the website from Rails to React. If you can code, consider volunteering, as we always need more help. The website is run mainly with Ruby on Rails and React.
            </p>
            <p>
              I have several 
              <a
                class="chakra-link css-1108m31"
                href="/projects"
              >
                web apps
              </a>
               based on cubes, primarily designed to help with learning and practising blindfolded solving -- and I intend to create more in the future. If you've recently learned to solve a cube blindfolded, or struggle with mistakes while practising, then you'll find them helpful.
            </p>
            <p>
              There are plenty of other tools out there made by other people. I don't have a comprehensive list handy, but here are a few notable sites:
            </p>
            <ul
              class="css-tu0njr"
              role="list"
            >
              <li
                class="css-0"
              >
                <a
                  class="chakra-link css-1108m31"
                  href="https://github.com/cubing/cubing.js/"
                  rel="noopener"
                  target="_blank"
                  to=""
                >
                  cubing.js
                </a>
                : a collection of javascript libraries
              </li>
              <li
                class="css-0"
              >
                <a
                  class="chakra-link css-1108m31"
                  href="https://alg.cubing.net/"
                  rel="noopener"
                  target="_blank"
                  to=""
                >
                  alg.cubing
                </a>
                : a 3-D cube simulator
              </li>
              <li
                class="css-0"
              >
                <a
                  class="chakra-link css-1108m31"
                  href="https://cstimer.net/"
                  rel="noopener"
                  target="_blank"
                  to=""
                >
                  csTimer
                </a>
                : a timer for timing solves
              </li>
            </ul>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-2"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-2"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              Reconstructions of some of my notable solves can be found 
              <a
                class="chakra-link css-1108m31"
                href="https://docs.google.com/spreadsheets/d/1PJ1k_EXqv_mafcMDRbrD5OAyBHie2z5PH5OkHLs8BtE/edit#gid=1878622909"
                rel="noopener"
                target="_blank"
                to=""
              >
                here
              </a>
              , including both regular and blindfolded solving for several different events. Note that you can apply existing filters to filter by event.
            </p>
            <p>
              Reconstructions of all my fewest moves attempts can be found here 
              <a
                class="chakra-link css-1108m31"
                href="https://docs.google.com/spreadsheets/d/1Ecdjl4S4IZD6x60x3U5rHXlfV8HIDMEbk2SEaiZruaE"
                rel="noopener"
                target="_blank"
                to=""
              >
                here
              </a>
              , but note that this is spreadsheet is a work-in-progress and not all my solves have been added yet. There are numerous pages of statistics.
            </p>
            <p>
              For more interesting reconstructions, of fast cubers, see 
              <a
                class="chakra-link css-1108m31"
                href="https://cubesolv.es"
                rel="noopener"
                target="_blank"
                to=""
              >
                cubesolv.es
              </a>
              .
            </p>
            <p>
              To make your own reconstructions, use 
              <a
                class="chakra-link css-1108m31"
                href="https://alg.cubing.net/"
                rel="noopener"
                target="_blank"
                to=""
              >
                alg.cubing
              </a>
              .
            </p>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-3"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-3"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              I have 
              <a
                class="chakra-link css-1108m31"
                href="/blog"
              >
                an article
              </a>
               explaining the core concepts behind blindfolded solving. Essentially, you memorize a sequence of letters where each letter is an instruction for what to do next, and you convert these letters to images/sentences/stories to remember them.
            </p>
            <p>
              I have 
              <a
                class="chakra-link css-1108m31"
                href="https://docs.google.com/spreadsheets/d/1Ml7Jl5Yeb9HeE14xO7j1Nc99sw08oiBjLbi_nYJk9rU/edit"
                rel="noopener"
                target="_blank"
                to=""
              >
                a spreadsheet
              </a>
               which lists the images I use for this, and the move sequences (also known as algorithms in the community) which I use to solve each pair of pieces. (I also use the images for memory sports, including memorizing numbers and playing cards.)
            </p>
            <p>
              My system is relatively unique in the blindfolded solving community in that it is based on phonemes and pronunciation rather than letters and spelling. Memory sports athletes usually use phonemes, as far as I understand, though I came up with the idea independently. It took literally years of experimenting to get my system to where it is today.
            </p>
            <p>
              There are many online resources for blindfolded solving in general, including 
              <a
                class="chakra-link css-1108m31"
                href="https://blddb.net/"
                rel="noopener"
                target="_blank"
                to=""
              >
                this database of algorithms
              </a>
              , 
              <a
                class="chakra-link css-1108m31"
                href="https://bestsiteever.ru/tables/"
                rel="noopener"
                target="_blank"
                to=""
              >
                this compilation of algorithm lists
              </a>
              , and 
              <a
                class="chakra-link css-1108m31"
                href="https://docs.google.com/spreadsheets/d/1Fi4xgUz5b23UXMlHq7Tt5C8Ak8-U3XdbeQ9Anw68BQc"
                rel="noopener"
                target="_blank"
                to=""
              >
                this collection of letter pair images
              </a>
              .
            </p>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-4"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-4"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              Together with 
              <a
                class="chakra-link css-1108m31"
                href="https://www.worldcubeassociation.org/persons/2016SIGG01"
                rel="noopener"
                target="_blank"
                to=""
              >
                Graham Siggins
              </a>
               I developed a slight optimization for multiple-blindfolded solving called Weak Swap. It is a generalization of 'the swap', a common method to handle parity efficiently in regular blindfolded solving.
            </p>
            <p>
              Graham pointed out a special case to me one day where he was able to reduce his algorithm count by one. I played around and realized that it could be generalized to apply to a large portion of scrambles, and together we nailed down what turned out to be a relatively simple method to incorporate it into every cube in an attempt. Graham has a 
              <a
                class="chakra-link css-1108m31"
                href="https://www.youtube.com/watch?v=MyeQkcsAzUE&ab_channel=GrahamSiggins"
                rel="noopener"
                target="_blank"
                to=""
              >
                sequence of videos
              </a>
               on the topic, and the calculations I did can be found on 
              <a
                class="chakra-link css-1108m31"
                href="https://github.com/kr-matthews/weak-swap"
                rel="noopener"
                target="_blank"
                to=""
              >
                GitHub
              </a>
              . I intend to eventually write a blog post about it.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
